test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java

```diff
@@ -31,12 +33,35 @@
 /**
  * Test appender that can be used to verify that certain events were logged correctly
  */
-public class MockLogAppender extends AbstractAppender {
+public class MockLogAppender {
```

Comment:
Q: since this is no longer an appender, could this break things in serverless? (if tests there use something like `Loggers.addAppender(mockAppender)`)

Reply:
It will "break" things, but I have the necessary changes ready.